src/components/AppSessionStateProvider.test.tsx

```diff
@@ -55,6 +55,60 @@ describe("AppSessionStateProvider", () => {
     expect(inspectCurrentSessionState().status).toEqual(AUTHENTICATED.status);
     expect(inspectCurrentSessionState().lang).toEqual(AUTHENTICATED.lang);
   });
+
+  it("should update sessionState when 1st event comes after render but before useEffect", () => {
+    const {
+      startMonitoringSessionState,
+      TestComponent,
+      inspectCurrentSessionState,
+      simulateSessionUpdate,
+    } = createTestContext();
+
+    expect(inspectCurrentSessionState()).toBeUndefined();
+
+    startMonitoringSessionState();
+
+    const ComponentToInjectCodeBeforeTestComponentUseEffect = () => {
+      simulateSessionUpdate(AUTHENTICATED);
+      return <></>;
+    };
+
+    render(
+      <>
+        <TestComponent />
+        <ComponentToInjectCodeBeforeTestComponentUseEffect />
+      </>
+    );
+
+    expect(inspectCurrentSessionState().status).toEqual(AUTHENTICATED.status);
+    expect(inspectCurrentSessionState().lang).toEqual(AUTHENTICATED.lang);
+  });
+
+  it("should update sessionState when 1st event comes before rendering", () => {
+    const {
+      startMonitoringSessionState,
+      TestComponent,
+      inspectCurrentSessionState,
+      simulateSessionUpdate,
+    } = createTestContext();
+
+    simulateSessionUpdate(AUTHENTICATED);
+
+    startMonitoringSessionState();
+
+    const ComponentToInjectCodeBeforeTestComponentUseEffect = () => {
+      expect(inspectCurrentSessionState().status).toEqual(AUTHENTICATED.status);
+      expect(inspectCurrentSessionState().lang).toEqual(AUTHENTICATED.lang);
+      return <></>;
+    };
+
+    render(
+      <>
+        <TestComponent />
+        <ComponentToInjectCodeBeforeTestComponentUseEffect />
+      </>
+    );
+  });
```

Comment:
This test is for testing this line of code:

![image](https://user-images.githubusercontent.com/1157864/179820381-d7e271a1-ed20-47eb-aa39-9542e5495941.png)

I also needed to use `ComponentToInjectCodeBeforeTestComponentUseEffect` because Testing Library executes the rendering and `useEffect` in a single step.